src/basic/time-util.h

```diff
@@ -211,6 +211,14 @@ static inline usec_t usec_sub_signed(usec_t timestamp, int64_t delta) {
         return usec_sub_unsigned(timestamp, (usec_t) delta);
 }
 
+static inline int usleep_safe(usec_t usec) {
+        /* usleep() takes useconds_t that is (typically?) uint32_t. Also, usleep() may only support the
+         * range [0, 1000000]. See usleep(3). Let's override usleep() with nanosleep(). */
+
+        // FIXME: use RET_NERRNO() macro here. Currently, this header cannot include errno-util.h.
+        return nanosleep(TIMESPEC_STORE(usec), NULL) < 0 ? -errno : 0;
```

Comment:
Please use `errno_or_else`. This doesn't matter much, since the return value is ignored, but it's better to be sure that we're not returning 0 on success here.

Reply:
As you said, currently the return value is never used. I'd like to keep it as is, at least now.
If `nanosleep()` fails with non-positive errno, then that should be glibc or kernel issue. I'd like to not introduce (unnecessary at least now) workaround.